gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +720,684 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+            {
+                /* Reading reference points */
+                fits_read_key(hFITS, TDOUBLE, "CRPIX1", &crpix1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRPIX2", &crpix2, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL1", &crval1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL2", &crval2, nullptr, &status);
+                /* Check for CDELT and PC matrix representation */
+                fits_read_key(hFITS, TDOUBLE, "CDELT1", &cdelt1, nullptr, &status);
+                if ( ! status ) {
+                    fits_read_key(hFITS, TDOUBLE, "CDELT2", &cdelt2, nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_1", &pc[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_2", &pc[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_1", &pc[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_2", &pc[3], nullptr, &status);
+                    cd[0] = cdelt1 * pc[0];
+                    cd[1] = cdelt1 * pc[1];
+                    cd[2] = cdelt2 * pc[2];
+                    cd[3] = cdelt2 * pc[3];
+                    status = 0;
+                } else {
+                    /* Look for CD matrix representation */
+                    fits_read_key(hFITS, TDOUBLE, "CD1_1", &cd[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD1_2", &cd[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_1", &cd[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_2", &cd[3], nullptr, &status);
+                }
+
+                double radfac = DEG2RAD * aRadius;
+
+                adfGeoTransform[1] = cd[0] * radfac;
+                adfGeoTransform[2] = cd[1] * radfac;
+                adfGeoTransform[4] = cd[2] * radfac;
+                adfGeoTransform[5] = - cd[3] * radfac ;
+                if ( crval1 > 180. ) {
+                    crval1 = crval1 - 180.;
+                }
+                adfGeoTransform[0] = crval1 * radfac - adfGeoTransform[1] *
+                                                    (crpix1-0.5);
+                adfGeoTransform[3] = crval2 * radfac - adfGeoTransform[5] *
+                                                    (crpix2-0.5);
+                bGeoTransformValid = true;
+            }
+
+            char* pstr = strrchr(ctype, '-') + 1;
+            char projstr[4];
+            strcpy(projstr, pstr);
+
+            /* Defining projection type
+               Following http://www.gdal.org/ogr__srs__api_8h.html (GDAL)
+               and http://www.aanda.org/component/article?access=bibcode&bibcode=&bibcode=2002A%2526A...395.1077CFUL (FITS)
+            */
+
+            /* Sinusoidal / SFL projection */
+            if( strcmp(projstr,"SFL" ) == 0 )
+            {
+                strncpy(projName, "Sinusoidal_", 12);
```

Comment:
please use the dedicated SetSinusoidal() method instead of SetProjection() + SetProjParm(). This applies to a number of the below other projections

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/228a04301fd7acdf29d9458dba0d7dd7e28c6951